java/src/jmri/jmrit/display/layoutEditor/LayoutTrack.java

```diff
@@ -217,5 +211,80 @@ public Rectangle2D getBounds() {
         return new Rectangle2D.Double();
     }
 
+    /**
+     * get the object connected to this track for this connection type
+     * @param connectionType
+     * @return the object connected to this layout track connected for the connection type
+     * @throws jmri.JmriException
+     */
+    public Object getConnection(int connectionType) /* throws jmri.JmriException */ {
+        Object result = null;
+        try {
+            switch (connectionType) {
+                case POS_POINT: {
+                    result = ((PositionablePoint)this).getConnection(connectionType);
+                    break;
+                }
+                case TURNOUT_A:
+                case TURNOUT_B:
+                case TURNOUT_C:
+                case TURNOUT_D: {
+                    result = ((LayoutTurnout)this).getConnection(connectionType);
+                    break;
+                }
+                case LEVEL_XING_A:
+                case LEVEL_XING_B:
+                case LEVEL_XING_C:
+                case LEVEL_XING_D: {
+                    result = ((LevelXing)this).getConnection(connectionType);
+                    break;
+                }
+
+                case TRACK: {
+                    result = ((TrackSegment)this).getConnection(connectionType);
+                    break;
+                }
+
+                case SLIP_A:
+                case SLIP_B:
+                case SLIP_C:
+                case SLIP_D: {
+                    result = ((LayoutSlip)this).getConnection(connectionType);
+                    break;
+                }
+                default: {
+                    if (connectionType >= TURNTABLE_RAY_OFFSET) {
+                        result = ((LayoutTurntable)this).getConnection(connectionType);
+                    } else {
+                        log.error("Invalid connection type " + connectionType); //I18IN
+                    }
+                    break;
+                }
+            }
+        } catch (Exception e) {
+            //exceptions make me throw up…
+            log.error("This error message, which nobody will ever see, shuts my IDE up.");
```

Comment:
Got it. This method now throws the exception also and callers have been re-written to catch them.